docs/architecture/pain_points.md

```diff
@@ -37,10 +37,11 @@ in batch mode.
 Does not support dynamic destinations.
 
 Does not support [`NestedValueProvider`] for destinations in streaming mode on
-Dataflow, which is needed to create templates that accept a mapping of document
-type to a predetermined number of destinations. This is because Dataflow moves
-the implementation into the shuffler to improve performance. Current workaround
-is to specify mapping at template creation time.
+Dataflow, which is needed to create classic templates that accept a mapping of
+document type to a predetermined number of destinations. This is because
+Dataflow moves the implementation into the shuffler to improve performance.
+Current workaround is to specify mapping at classic template creation time, or
+use flex templates.
```

Comment:
```suggestion
use Flex Templates.
```

I'm being a bit pedantic here, but GCP docs consistently use title case for Flex Templates and I do find it a bit confusing to see "flex template" which readers might think is referring to some generally known concept rather than a Dataflow-specific product feature.